### ISSUE TITLE
New rule no-method-to-bool-case

### ISSUE THREAD

**Issue by nkt** [eslint/eslint#7557]
This is very common typo, when you checks method instead of method call result.

**What category of rule is this? (place an "X" next to just one item)**

 -  [x] Warns about a potential error

**Provide 2-3 code examples that this rule will warn about:**

```js
isFocused(): boolean {
}

focus() {
  if (!this.isFocused) {
    this.input.focus()
  }
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Because this rule would be helpful in any JS project.

**Comment by platinumazure** [comment]
The difficulty with this proposal is that it's really hard to tell, in a static analyzer, what actually is a method (vs a boolean, vs uninitialized until provided via constructor, vs anything else). So I'm not sure how feasible this is.

Even for methods on the same object, since it's not always clear what value `this` has, we wouldn't be able to reliably detect those cases necessarily.

**Comment by nkt** [comment]
I guess first and simple implementation should collect all class methods and check a statement using collected results.
Smarter implementation should collect methods from parent classes, but I don't think this is really needed.

**Comment by platinumazure** [comment]
I'm :-1: on including this in core due to value/maintenance cost ratio
being too low. But we'll see what the rest of the team thinks. If the team
agrees, please note that this can always be implemented in a plugin.

**Comment by nzakas** [comment]
I agree. This isn't a good candidate for a core rule, but you you can certainly pursue a custom rule if it's important to you.

**Comment by ilyavolodin** [comment]
I also agree. This is too narrow of a scope for a core issue. This is also a browser specific rule, and while we have some of those, we don't have too many. This should go into a custom plugin.
@nkt Thanks for the suggestion. But it looks like the team feels that this is better fit for a plugin. If you need help creating one, please stop by our [Gitter](https://gitter.im/eslint/eslint) chatroom. Closing this issue.
